feature-libs/product-configurator/rulebased/components/attribute/types/input-field/configurator-attribute-input-field.component.ts

```diff
@@ -47,13 +86,15 @@ export class ConfiguratorAttributeInputFieldComponent
     const event: ConfigFormUpdateEvent = {
       ownerKey: this.ownerKey,
       changedAttribute: {
-        name: this.attribute.name,
+        ...this.attribute,
         userInput: this.attributeInputForm.value,
-        uiType: this.attribute.uiType,
-        groupId: this.attribute.groupId,
       },
     };
 
     this.inputChange.emit(event);
   }
+
+  ngOnDestroy() {
+    this.sub.unsubscribe();
```

Comment:
To make it safer:
```
if (this.sub) {
      this.sub.unsubscribe();
}
```

Reply:
picked

Reply:
FYI a simpler alternative is just `this.sub?.unsubscribe();`